client_internals_it_test.go

```diff
@@ -208,3 +191,129 @@ func TestClusterID(t *testing.T) {
 		})
 	})
 }
+
+func TestClientInternal_ClusterID(t *testing.T) {
+	tc := it.StartNewClusterWithOptions("ci-cluster-id", 55701, 1)
+	ctx := context.Background()
+	client := it.MustClient(hz.StartNewClientWithConfig(ctx, tc.DefaultConfig()))
+	defer client.Shutdown(ctx)
+	ci := hz.NewClientInternal(client)
+	assert.NotEqual(t, types.UUID{}, ci.ClusterID())
+	tc.Shutdown()
+	assert.Equal(t, types.UUID{}, ci.ClusterID())
+}
+
+func TestClientInternal_InvokeOnRandomTarget(t *testing.T) {
+	clientInternalTester(t, "ci-invoke-random", func(t *testing.T, ci *hz.ClientInternal) {
+		ctx := context.Background()
+		t.Run("without handler", func(t *testing.T) {
+			if _, err := ci.InvokeOnRandomTarget(ctx, codec.EncodeClientPingRequest(), nil); err != nil {
+				t.Fatal(err)
+			}
+		})
+		t.Run("with handler", func(t *testing.T) {
+			invoked := int32(0)
+			opts := &hz.InvokeOptions{
+				Handler: func(clientMessage *hz.ClientMessage) {
+					atomic.StoreInt32(&invoked, 1)
+				},
+			}
+			req := codec.EncodeMapAddEntryListenerRequest("foo", true, int32(hz.EntryAdded), false)
+			if _, err := ci.InvokeOnRandomTarget(ctx, req, opts); err != nil {
+				t.Fatal(err)
+			}
+			m, err := ci.Client().GetMap(ctx, "foo")
+			if err != nil {
+				t.Fatal(err)
+			}
+			if err := m.Set(ctx, "key", "value"); err != nil {
+				t.Fatal(err)
+			}
+			it.Eventually(t, func() bool {
+				return atomic.LoadInt32(&invoked) == 1
+			})
+		})
+
+	})
+}
+
+func TestClientInternal_InvokeOnPartition(t *testing.T) {
+	clientInternalTester(t, "ci-invoke-partition", func(t *testing.T, ci *hz.ClientInternal) {
+		if _, err := ci.InvokeOnPartition(context.Background(), codec.EncodeClientPingRequest(), 1, nil); err != nil {
+			t.Fatal(err)
+		}
+	})
+}
+
+func TestClientInternal_InvokeOnKey(t *testing.T) {
+	clientInternalTester(t, "ci-invoke-key", func(t *testing.T, ci *hz.ClientInternal) {
+		keyData, err := ci.EncodeData("foo")
+		if err != nil {
+			t.Fatal(err)
+		}
+		if _, err := ci.InvokeOnKey(context.Background(), codec.EncodeClientPingRequest(), keyData, nil); err != nil {
```

Comment:
I think it would be great If we can assert some fields of the response
for the tests:
```
TestClientInternal_InvokeOnPartition
TestClientInternal_InvokeOnMember
TestClientInternal_InvokeOnRandomTarget

```

Reply:
Ping message doesn't return a decodable response. I've replaced it MC GetTimedMembers codec at: https://github.com/hazelcast/hazelcast-go-client/pull/754/commits/dfbbe32e0800e6136af7ddcee3dafb94e0bef552